src/components/collectible-icon/collectible-icon.tsx

```diff
@@ -11,92 +17,105 @@ import {
-  return strategy[0];
-};
-
 export const CollectibleIcon: FC<CollectibleIconProps> = ({
   collectible,
   size,
-  iconSize = CollectibleIconSize.SMALL
+  iconSize = CollectibleIconSize.SMALL,
+  mime,
+  objktArtifact,
+  setScrollEnabled
 }) => {
+  const [isLoading, setIsLoading] = useState(true);
+
+  const isBigIcon = iconSize === CollectibleIconSize.BIG;
   const styles = useCollectibleIconStyles();
-  const actualLoadingStrategy =
-    iconSize === CollectibleIconSize.SMALL ? collectibleThumbnailLoadStrategy : collectibleBigLoadStrategy;
-  const [isLoadingFailed, setIsLoadingFailed] = useState(
-    actualLoadingStrategy.reduce<Record<string, boolean>>((acc, cur) => ({ ...acc, [cur.type]: false }), {})
-  );
-  const assetSlug = `${collectible?.address}_${collectible?.id}`;
+  const assetSlug = `${collectible.address}_${collectible.id}`;
+
+  const initialFallback = useMemo(() => {
+    if (isDefined(collectible.artifactUri) && isBigIcon) {
+      return formatCollectibleObjktArtifactUri(collectible.artifactUri);
+    }
+
+    return formatCollectibleObjktMediumUri(assetSlug);
+  }, []);
 
-  const imageRequestObject = { ...collectible, assetSlug };
-  const currentFallback = getFirstFallback(actualLoadingStrategy, isLoadingFailed, imageRequestObject);
-  const imageSrc = currentFallback.uri(imageRequestObject[currentFallback.field] ?? assetSlug);
+  const [isAnimatedRenderedOnce, setIsAnimatedRenderedOnce] = useState(false);
+  const [currentFallback, setCurrentFallback] = useState(initialFallback);
+
+  const handleError = () =>
+    void setCurrentFallback(
+      formatImgUri(isDefined(collectible.artifactUri) ? collectible.artifactUri : collectible.thumbnailUri, 'medium')
+    );
+
+  const handleLoadEnd = () => setIsLoading(false);
+
+  const icon = useMemo(() => {
+    if (!isAnimatedRenderedOnce && isDefined(objktArtifact) && isBigIcon) {
+      if (isImgUriDataUri(objktArtifact)) {
+        setIsAnimatedRenderedOnce(true);
+
+        return (
+          <AnimatedSvg style={styles.image} dataUri={objktArtifact} onError={handleError} onLoadEnd={handleLoadEnd} />
+        );
+      }
+
+      if (mime === NonStaticMimeTypes.MODEL) {
+        setIsAnimatedRenderedOnce(true);
+
+        return (
+          <SimpleModelView
+            style={styles.image}
+            uri={formatCollectibleObjktArtifactUri(objktArtifact)}
+            onError={handleError}
+            onLoadEnd={handleLoadEnd}
+            setScrollEnabled={setScrollEnabled}
+          />
+        );
+      }
+
+      if (mime === NonStaticMimeTypes.VIDEO) {
+        setIsAnimatedRenderedOnce(true);
+
+        return (
+          <SimplePlayer
+            uri={formatCollectibleObjktArtifactUri(objktArtifact)}
+            size={size}
+            style={styles.image}
+            onError={handleError}
+            onLoad={handleLoadEnd}
+          />
+        );
+      }
+
+      if (mime === NonStaticMimeTypes.AUDIO) {
+        setIsAnimatedRenderedOnce(true);
```

Comment:
`setIsAnimatedRenderedOnce(true);` - you used it in all cases, can we move it to the first condition?

Reply:
first condition doesnt guarantee that animated component was rendered once